Complete front-end setup and integration

SVijayB opened the pull request:
<!-- ⚠️⚠️ Do Not Delete These Comments. ⚠️⚠️ -->
<!-- Please read these comments/instructions carefully and do accordingly  -->
<!-- Read our contributing document : https://github.com/SVijayB/Quizzatron/blob/master/.github/CONTRIBUTING.md -->
<!-- Read our Rules of Conduct: https://github.com/SVijayB/Quizzatron/blob/master/.github/CODE_OF_CONDUCT.md -->
<!--- Provide a general summary of your changes in the Title above -->

### Description

Completed setting up the frontend and integration with backend API.
I have also migrated the code from gemini old SDK to the new one. ([Docs](https://ai.google.dev/gemini-api/docs/migrate)).
Validated a lot of the model responses and updated the final json format accordingly to pass for several of them.
Frontend adds more layers of complexity and gamification to the quiz.
Each question has 10 seconds to answer, how quickly you answer it will add points to result page.

### Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
Needed to setup a frontend to interact with the API and test the code.
It's fully functional now.
To install and setup front-end, make sure to have node and npm installed.
Then cd `frontend/` and run `npm i`. 
Once the packages are installed, run `npm run dev`.

### Screenshots (if any)

![image](https://github.com/user-attachments/assets/1629c753-f706-4b44-a63b-ad99053de799)

![image](https://github.com/user-attachments/assets/ef6bc40d-3e2b-4ca8-b32d-fd51e5bc35b2)

![image](https://github.com/user-attachments/assets/f5ec5bb1-1126-4fbf-95c0-f37fe5e70fb3)



### Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] My code follows the code style of this project.
-   [x] My change requires a change to the documentation.
-   [x] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.

<!-- Before submitting, click on the preview tab to check your work so far-->

akshayravi13 reviewed (commented):
Need to setup CORS for the connection of backend and frontend on a public server.

SVijayB commented:
I've updated the CORS policy for public servers. 
I've also updated the front-end code to match the current codespaces generated backend URL.
You can test out the public front-end deployment here: https://silver-capybara-vrp6rqj6jxvfwrvp-8080.app.github.dev/

akshayravi13 reviewed (approved):
LGTM! Everything works well. Tested out the netlify app as well. Image rendering is also working on public site. We're good to go.
